nav2_smoother/src/nav2_smoother.cpp

```diff
@@ -49,7 +49,6 @@ SmootherServer::SmootherServer()
     "footprint_topic",
     rclcpp::ParameterValue(
       std::string("global_costmap/published_footprint")));
-  declare_parameter("global_frame", rclcpp::ParameterValue(std::string("map")));
```

Comment:
This parameter is removed, that's OK, we just have to remove it from the config guide documentation. But before we do that, do you think the plugins will want to know this? If so, maybe we keep it in knowing that it'll be used later.

Reply:
You mean that the parameter would remain declared so that plugins could use it? Well, unlike in recovery server, I don't think a usual smoother would get use of it. Smoothers get all input data (path poses, costmap) in one frame (+ footprint in robot frame but there's no further need for tf lookup, it can be transformed directly to path poses), work with them in one frame and return the results in the same frame. If there was a special smoother (e.g. which considers other robots' footprints), it can still declare its own parameter (or get the global frame from the header of an oriented footprint).

BTW, when I found this PR unmerged I took liberty to do one final minor optimization that came to mind, hopefully it's OK.